UniConnectApp/app/src/main/java/com/example/uniconnect/MainViewModel.kt

```diff
@@ -0,0 +1,20 @@
+package com.example.uniconnect
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.uniconnect.dto.University
+import com.example.uniconnect.service.UniversityService
+import kotlinx.coroutines.launch
+
+class MainViewModel : ViewModel() {
+    var universities: MutableLiveData<List<University>> = MutableLiveData<List<University>>()
+    var UniversityService: UniversityService = UniversityService()
+
+    fun fetchUniversities() {
+        viewModelScope.launch {
+            var innerUniversity = UniversityService.fetchUniversities()
+            universities.postValue(innerUniversity)
+        }
+    }
+}
```

Comment:
Added MainViewModel as it didn't had it.

Reply:
This looks like an incomplete feature.
Are you using these anywhere?  I don't see them used.
If not in use, remove them.  Unused code is clutter, and bad documentation, which increases technical debt.
Only complete features, which meet the team's Definition of Done, should be merged to Main/Master.